vscode-extension/src/devbox.ts

```diff
@@ -0,0 +1,61 @@
+import { window, workspace, commands, ProgressLocation, Uri } from 'vscode';
+import { spawn, spawnSync } from 'node:child_process';
+
+
+interface Message {
+    status: string
+}
+export async function devboxReopen() {
+    await window.withProgress({
+        location: ProgressLocation.Notification,
+        title: "Setting up your Devbox environment. Please don't close vscode.",
+        cancellable: true
+    },
+        async (progress, token) => {
+            token.onCancellationRequested(() => {
+                console.log("User canceled the long running operation");
+            });
+
+            const p = new Promise<void>(async (resolve, reject) => {
+
+                if (workspace.workspaceFolders) {
+                    const workingDir = workspace.workspaceFolders[0].uri;
+                    const dotdevbox = Uri.joinPath(workingDir, '/.devbox');
+                    try {
+                        // check if .devbox exists
+                        await workspace.fs.stat(dotdevbox);
+                    } catch (error) {
+                        //.devbox doesn't exist
+                        // running devbox shellenv to create it
+                        spawnSync('devbox', ['shellenv'], {
+                            cwd: workingDir.path
+                        });
+                    }
+                    // change this to absolute path to custom compiled devbox when testing
+                    const devbox = 'devbox';
```

Comment:
this would have to be something else for prod?

Reply:
@LucilleH No, for testing locally on a compiled devbox this would have to point to the path to the compiled devbox.
This comment is mostly a note-to-self for future changes.
For example, if I change something in CLI and want to see how it interacts with the extension, I should change this line to `/Users/mohsen/projects/devbox/dist/devbox` so that my changes in compiled devbox CLI gets used instead of the installed devbox. Hope that makes sense.

Reply:
Ah, `To use a custom compiled devbox when testing, change this to an absolute path.` lol